### ISSUE TITLE
predbat.battery_hours_left not updated and stuck to 48h

### ISSUE THREAD

**Issue by marcoaddario** [springfall2008/batpred#959]
**Describe the bug**
battery_hours_left is stuck to 48 hours since the night charge. It already happened with a previous version of predbat and I fixed it restarting appdaemon. after that the value went up to 48h and did not update. 

**Expected behavior**
battery_hours_left should be evaluated and display the predicted time remaining on battery

**Predbat version**

currently on 7.16.15 but I got the same with the previous (at least 2) versions

**Environment details**

- Inverter and battery setup
  Solis Modbus
- Standard HAOS installer or Docker
  standard

**Screenshots**
<img width="1376" alt="image" src="https://github.com/springfall2008/batpred/assets/3762730/dbe3528c-873a-4f82-8a8e-bc4b04f97865">

**Log file**

the notable events are:
2024-04-12 23:00:02.100769 INFO pred_bat: Battery has 3.5 hours left - now at 3.9

2024-04-12 23:00:25.977999 INFO pred_bat: Battery has 3.5 hours left - now at 3.9

2024-04-12 23:05:02.105113 INFO pred_bat: Battery has 48.0 hours left - now at 3.83

and it stayed to 48.0 since then:
```
2024-04-12 22:10:02.129282 INFO pred_bat: Battery has 4.0 hours left - now at 4.54
2024-04-12 22:10:05.456293 INFO pred_bat: Battery has 4.0 hours left - now at 4.54
2024-04-12 22:15:02.396427 INFO pred_bat: Battery has 3.92 hours left - now at 4.47
2024-04-12 22:20:02.473727 INFO pred_bat: Battery has 3.83 hours left - now at 4.4
2024-04-12 22:20:06.019344 INFO pred_bat: Battery has 3.83 hours left - now at 4.4
2024-04-12 22:25:01.867425 INFO pred_bat: Battery has 3.67 hours left - now at 4.33
2024-04-12 22:30:01.977237 INFO pred_bat: Battery has 3.58 hours left - now at 4.26
2024-04-12 22:30:05.353513 INFO pred_bat: Battery has 3.58 hours left - now at 4.26
2024-04-12 22:35:02.430481 INFO pred_bat: Battery has 3.5 hours left - now at 4.19
2024-04-12 22:40:02.133889 INFO pred_bat: Battery has 3.33 hours left - now at 4.12
2024-04-12 22:40:05.462428 INFO pred_bat: Battery has 3.33 hours left - now at 4.12
2024-04-12 22:45:02.054342 INFO pred_bat: Battery has 3.33 hours left - now at 4.12
2024-04-12 22:50:02.037761 INFO pred_bat: Battery has 3.25 hours left - now at 4.05
2024-04-12 22:50:05.493454 INFO pred_bat: Battery has 3.25 hours left - now at 4.05
2024-04-12 22:55:02.354876 INFO pred_bat: Battery has 3.25 hours left - now at 3.98
2024-04-12 23:00:02.100769 INFO pred_bat: Battery has 3.5 hours left - now at 3.9
2024-04-12 23:00:25.977999 INFO pred_bat: Battery has 3.5 hours left - now at 3.9
2024-04-12 23:05:02.105113 INFO pred_bat: Battery has 48.0 hours left - now at 3.83
2024-04-12 23:10:02.238417 INFO pred_bat: Battery has 48.0 hours left - now at 3.76
2024-04-12 23:10:05.559830 INFO pred_bat: Battery has 48.0 hours left - now at 3.76
2024-04-12 23:15:01.929021 INFO pred_bat: Battery has 48.0 hours left - now at 3.69
2024-04-12 23:20:02.193269 INFO pred_bat: Battery has 48.0 hours left - now at 3.62
2024-04-12 23:20:05.704604 INFO pred_bat: Battery has 48.0 hours left - now at 3.62
2024-04-12 23:25:01.965372 INFO pred_bat: Battery has 48.0 hours left - now at 3.55
2024-04-12 23:30:02.127640 INFO pred_bat: Battery has 48.0 hours left - now at 3.55
2024-04-12 23:30:11.650823 INFO pred_bat: Battery has 48.0 hours left - now at 3.55
2024-04-12 23:35:02.032932 INFO pred_bat: Battery has 48.0 hours left - now at 3.76
2024-04-12 23:40:02.041939 INFO pred_bat: Battery has 48.0 hours left - now at 4.05
2024-04-12 23:40:05.477991 INFO pred_bat: Battery has 48.0 hours left - now at 4.05
2024-04-12 23:45:02.010641 INFO pred_bat: Battery has 48.0 hours left - now at 4.33
2024-04-12 23:50:02.057765 INFO pred_bat: Battery has 48.0 hours left - now at 4.62
2024-04-12 23:50:05.639018 INFO pred_bat: Battery has 48.0 hours left - now at 4.62
2024-04-12 23:55:02.403420 INFO pred_bat: Battery has 48.0 hours left - now at 4.9
2024-04-13 00:00:04.744846 INFO pred_bat: Battery has 48.0 hours left - now at 5.11
2024-04-13 00:05:02.386440 INFO pred_bat: Battery has 48.0 hours left - now at 5.4
2024-04-13 00:10:02.005811 INFO pred_bat: Battery has 48.0 hours left - now at 5.68
2024-04-13 00:10:05.540850 INFO pred_bat: Battery has 48.0 hours left - now at 5.68
2024-04-13 00:15:02.476782 INFO pred_bat: Battery has 48.0 hours left - now at 5.96
2024-04-13 00:20:01.845418 INFO pred_bat: Battery has 48.0 hours left - now at 6.25
2024-04-13 00:20:05.546917 INFO pred_bat: Battery has 48.0 hours left - now at 6.25
2024-04-13 00:25:01.911499 INFO pred_bat: Battery has 48.0 hours left - now at 6.32
2024-04-13 00:30:01.995280 INFO pred_bat: Battery has 48.0 hours left - now at 6.32
2024-04-13 00:30:05.435997 INFO pred_bat: Battery has 48.0 hours left - now at 6.32
2024-04-13 00:35:01.992953 INFO pred_bat: Battery has 48.0 hours left - now at 6.75
2024-04-13 00:40:02.114912 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 00:40:05.425743 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 00:45:02.429548 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 00:50:01.789121 INFO pred_bat: Battery has 48.0 hours left - now at 7.1
2024-04-13 00:50:07.297373 INFO pred_bat: Battery has 48.0 hours left - now at 7.1
2024-04-13 00:55:02.393467 INFO pred_bat: Battery has 48.0 hours left - now at 7.1
2024-04-13 01:00:01.915267 INFO pred_bat: Battery has 48.0 hours left - now at 7.1
2024-04-13 01:00:05.363366 INFO pred_bat: Battery has 48.0 hours left - now at 7.1
2024-04-13 01:05:02.235432 INFO pred_bat: Battery has 48.0 hours left - now at 7.1
2024-04-13 01:10:02.350712 INFO pred_bat: Battery has 48.0 hours left - now at 7.1
2024-04-13 01:10:05.571110 INFO pred_bat: Battery has 48.0 hours left - now at 7.1
2024-04-13 01:15:01.979480 INFO pred_bat: Battery has 48.0 hours left - now at 7.1
2024-04-13 01:20:01.805482 INFO pred_bat: Battery has 48.0 hours left - now at 7.1
2024-04-13 01:20:05.087940 INFO pred_bat: Battery has 48.0 hours left - now at 7.1
2024-04-13 01:25:01.921430 INFO pred_bat: Battery has 48.0 hours left - now at 7.1
2024-04-13 01:30:01.861307 INFO pred_bat: Battery has 48.0 hours left - now at 7.1
2024-04-13 01:30:05.156714 INFO pred_bat: Battery has 48.0 hours left - now at 7.1
2024-04-13 01:35:01.763232 INFO pred_bat: Battery has 48.0 hours left - now at 7.1
2024-04-13 01:40:02.138527 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 01:40:07.475080 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 01:45:01.952660 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 01:50:01.917210 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 01:50:05.157944 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 01:55:01.954870 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 02:00:02.100540 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 02:00:05.502476 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 02:05:01.950268 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 02:10:02.447676 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 02:10:05.614546 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 02:15:01.918610 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 02:20:01.880435 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 02:20:05.136520 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 02:25:02.330799 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 02:30:01.988452 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 02:30:05.136033 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 02:35:02.165884 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 02:40:02.498767 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 02:40:05.606132 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 02:45:01.925255 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 02:50:01.920305 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 02:50:05.102430 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 02:55:01.910390 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 03:00:01.881751 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 03:00:05.190422 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 03:05:02.504689 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 03:10:01.982379 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 03:10:05.138179 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 03:15:01.942287 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 03:20:01.919851 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 03:20:05.070666 INFO pred_bat: Battery has 48.0 hours left - now at 7.03
2024-04-13 03:25:02.292807 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 03:30:01.921564 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 03:30:05.111903 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 03:35:02.039689 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 03:40:02.015984 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 03:40:05.244849 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 03:45:02.254990 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 03:50:01.901640 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 03:50:04.960306 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 03:55:01.919692 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 04:00:01.873048 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 04:00:05.146046 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 04:05:01.925503 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 04:10:01.793657 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 04:10:04.989752 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 04:15:02.365374 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 04:20:01.903208 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 04:20:05.001338 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 04:25:02.236375 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 04:30:01.768139 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 04:30:04.750863 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 04:35:02.239990 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 04:40:01.971922 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 04:40:05.068008 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 04:45:02.232490 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 04:50:02.015685 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 04:50:05.057560 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 04:55:02.176244 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 05:00:01.905940 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 05:00:05.045793 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 05:05:01.989253 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 05:10:01.864983 INFO pred_bat: Battery has 48.0 hours left - now at 6.89
2024-04-13 05:10:04.833660 INFO pred_bat: Battery has 48.0 hours left - now at 6.89
2024-04-13 05:15:02.025616 INFO pred_bat: Battery has 48.0 hours left - now at 6.89
2024-04-13 05:20:01.908156 INFO pred_bat: Battery has 48.0 hours left - now at 7.1
2024-04-13 05:20:06.845972 INFO pred_bat: Battery has 48.0 hours left - now at 7.1
2024-04-13 05:25:01.954746 INFO pred_bat: Battery has 48.0 hours left - now at 7.1
2024-04-13 05:30:01.914692 INFO pred_bat: Battery has 48.0 hours left - now at 7.1
2024-04-13 05:30:11.663433 INFO pred_bat: Battery has 48.0 hours left - now at 7.1
2024-04-13 05:35:01.923899 INFO pred_bat: Battery has 48.0 hours left - now at 6.96
2024-04-13 05:40:01.998361 INFO pred_bat: Battery has 48.0 hours left - now at 6.89
2024-04-13 05:40:05.410156 INFO pred_bat: Battery has 48.0 hours left - now at 6.89
2024-04-13 05:45:01.837255 INFO pred_bat: Battery has 48.0 hours left - now at 6.89
2024-04-13 05:50:01.998234 INFO pred_bat: Battery has 48.0 hours left - now at 6.82
2024-04-13 05:50:05.401078 INFO pred_bat: Battery has 48.0 hours left - now at 6.82
2024-04-13 05:55:01.991057 INFO pred_bat: Battery has 48.0 hours left - now at 6.82
2024-04-13 06:00:01.839618 INFO pred_bat: Battery has 48.0 hours left - now at 6.75
2024-04-13 06:00:05.450624 INFO pred_bat: Battery has 48.0 hours left - now at 6.75
2024-04-13 06:05:01.934318 INFO pred_bat: Battery has 48.0 hours left - now at 6.75
2024-04-13 06:10:01.900105 INFO pred_bat: Battery has 48.0 hours left - now at 6.67
2024-04-13 06:10:05.385976 INFO pred_bat: Battery has 48.0 hours left - now at 6.67
2024-04-13 06:15:01.824344 INFO pred_bat: Battery has 48.0 hours left - now at 6.67
2024-04-13 06:20:02.117986 INFO pred_bat: Battery has 48.0 hours left - now at 6.67
2024-04-13 06:20:05.519817 INFO pred_bat: Battery has 48.0 hours left - now at 6.67
2024-04-13 06:25:01.769010 INFO pred_bat: Battery has 48.0 hours left - now at 6.6
2024-04-13 06:30:01.932924 INFO pred_bat: Battery has 48.0 hours left - now at 6.6
2024-04-13 06:30:05.325007 INFO pred_bat: Battery has 48.0 hours left - now at 6.6
2024-04-13 06:35:01.900210 INFO pred_bat: Battery has 48.0 hours left - now at 6.53
2024-04-13 06:40:02.116575 INFO pred_bat: Battery has 48.0 hours left - now at 6.53
2024-04-13 06:40:05.440440 INFO pred_bat: Battery has 48.0 hours left - now at 6.53
2024-04-13 06:45:01.898596 INFO pred_bat: Battery has 48.0 hours left - now at 6.46
2024-04-13 06:50:02.131876 INFO pred_bat: Battery has 48.0 hours left - now at 6.46
2024-04-13 06:50:05.427181 INFO pred_bat: Battery has 48.0 hours left - now at 6.46
2024-04-13 06:55:02.447625 INFO pred_bat: Battery has 48.0 hours left - now at 6.39
[...]

```
full log in attachment
[log.txt](https://github.com/springfall2008/batpred/files/14977666/log.txt)

**Comment by springfall2008** [comment]
Battery hours left will remain at 48 if you have an upcoming charge slot, it will only drop below that if you are predicted to run out of battery before the next charge.

**Comment by marcoaddario** [comment]
Is this behavior changed recently?
this is the battery hours_left graph from the 1st of April (but unfortunately I cannot remember to which version I have upgraded around the 8th).
the 2 ramps on 12th and 17th are after I restarted appdaemon.
My charge plan is always more or less the same: charges during night following the intelligent octopus tariff

<img width="1387" alt="image" src="https://github.com/springfall2008/batpred/assets/3762730/7741023c-cd29-49d8-a970-8fa70081de92">
